### PR TITLE
[#1346] Remove event bus from credential management endpoint

### PR DESCRIPTION
**Pull request by jbtrystram**

### CODE REVIEW

File: service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+     * Returns an empty String as this implementation does not use event bus.
+     */
+    @Override
+    protected String getEventBusAddress() {
+        return "";
+    }
+
+    @Override
+    public String getName() {
+        return String.format("%s/%s",
```
Comment (dejanb) [inline service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java]: We moved this to the static field in other implementations.

File: service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+        final Span span = Util.newChildSpan(SPAN_NAME_UPDATE_CREDENTIALS, TracingHandler.serverSpanContext(ctx), tracer,
+                getClass().getSimpleName());
+
+        final JsonArray credentials = ctx.get(KEY_REQUEST_BODY);
+
+        final String deviceId = getDeviceIdParam(ctx);
+        final Optional<String> resourceVersion = Optional.ofNullable(ctx.get(KEY_RESOURCE_VERSION));
+        final String tenantId = getTenantParam(ctx);
+
+        if (tenantId == null) {
```
Comment (dejanb) [inline service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java]: Take a look at how `Util.getRequestParam()` is used in tenant implementation. We should be consistent here and change device endpoint as well.

File: service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+            logger.debug("missing payload");
+            TracingHelper.logError(span, "missing payload");
+            ctx.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, "missing payload"));
+        } else {
+
+            try {
+                decodeCredentials(credentials);
+            } catch (final IllegalArgumentException | IllegalStateException e) {
+                logger.debug("Error parsing credentials");
+                ctx.fail(new ClientErrorException(HttpURLConnection.HTTP_BAD_REQUEST, e));
```
Comment (dejanb) [inline service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java]: We need to finish response and span here (and return?)

File: service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+        }
+    }
+
+    private void getCredentialsForDevice(final RoutingContext ctx) {
+
+        final Span span = Util.newChildSpan(SPAN_NAME_GET_CREDENTIALS, TracingHandler.serverSpanContext(ctx), tracer,
+                getClass().getSimpleName());
+
+        // mandatory params
+        final String tenantId = getTenantParam(ctx);
```
Comment (dejanb) [inline service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java]: Again, see if you can use getRequestParam()

File: service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+
+    /**
+     * Decode a credential, based on the provided type.
+     *
+     * @param type The type of the secret. Will never be {@code null}.
+     * @param object The JSON object to decode. Will never be {@code null}.
+     * @return The decoded secret.
+     * @throws IllegalArgumentException If the credential object is invalid.
+     */
+    protected CommonCredential decodeCredential(final String type, final JsonObject object) {
```
Comment (dejanb) [inline service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java]: Formatting  (tabs) are not right

File: services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/AutowiredCredentialsManagementAdapter.java [hunk 6]
```diff
@@ -28,6 +28,7 @@
  */
 @Component
 @ConditionalOnBean(CredentialsManagementService.class)
+@Deprecated(forRemoval = true)
```
Comment (dejanb) [inline services/device-registry/src/main/java/org/eclipse/hono/deviceregistry/AutowiredCredentialsManagementAdapter.java]: Add a deprecation comment

File: service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+            response.setStatusCode(status);
+            switch (status) {
+            case HttpURLConnection.HTTP_OK:
+                final JsonArray credentialsArray = new JsonArray();
+                for (final CommonCredential credential : operationResult.getPayload()) {
+                    credentialsArray.add(JsonObject.mapFrom(credential));
+                }
+                operationResult.getResourceVersion().ifPresent(v -> response.putHeader(HttpHeaders.ETAG, v));
+                HttpUtils.setResponseBody(response, credentialsArray);
+                response.end();
```
Comment (dejanb) [inline service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java]: This seems unnecessary.

File: service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+        final Optional<String> resourceVersion = Optional.ofNullable(getRequestParam(KEY_RESOURCE_VERSION, ctx, span, true));
+
+        if (credentials == null) {
+            final String msg = String.format("Missing request payload");
+            logger.debug(msg);
+            TracingHelper.logError(span, msg);
+            HttpUtils.badRequest(ctx, msg);
+            Tags.HTTP_STATUS.set(span, HttpURLConnection.HTTP_BAD_REQUEST);
+            span.finish();
+            return;
```
Comment (jbtrystram) [inline service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java]: I feel like I could extract this as `getMandatoryPayload()` in `AbstractHttpEndpoint`. @dejanb WDYT ?

Comment (dejanb) [reply service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java]: Yeah, looks like it

File: service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java [hunk 9]
```diff
@@ -188,12 +188,7 @@ protected final void extractOptionalJsonPayload(final RoutingContext ctx) {
      */
     protected final void extractRequiredJsonArrayPayload(final RoutingContext ctx) {
         extractRequiredJson(ctx, body -> {
-            final var payload = body.getBodyAsJsonArray();
-            if (payload != null && payload.getList() == null) {
-                // work around eclipse-vertx/vert.x#2993
-                return null;
-            }
-            return payload;
+            return body.getBodyAsJsonArray();
```
Comment (dejanb) [inline service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java]: Are we sure we don't need this anymore?

Comment (jbtrystram) [reply service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java]: the bug is fixed : https://github.com/eclipse-vertx/vert.x/issues/2993
and hono updated to a more recent version of vertx (3.8.4)

Comment (dejanb) [reply service-base/src/main/java/org/eclipse/hono/service/http/AbstractHttpEndpoint.java]: OK. Great.

File: service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+
+        logger.debug("getCredentialsForDevice [tenant: {}, device-id: {}]]", tenantId, deviceId);
+        final Promise<OperationResult<List<CommonCredential>>> result = Promise.promise();
+
+        result.future().setHandler(handler -> {
+            final OperationResult<List<CommonCredential>> operationResult = handler.result();
+            final int status = operationResult.getStatus();
+            response.setStatusCode(status);
+            switch (status) {
+            case HttpURLConnection.HTTP_OK:
```
Comment (dejanb) [inline service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java]: case should be indented.

**Pull request merged**